misc/docerina/src/main/java/org/ballerinalang/docgen/Generator.java

```diff
@@ -453,24 +447,28 @@ private static Field getVariable(BLangFunction functionNode, BLangVariable param
      * @param recordType ballerina record node.
      * @return documentation of the record.
      */
-    public static RecordDoc createDocForType(BLangRecordTypeNode recordType, String structName) {
+    private static RecordDoc createDocForType(BLangTypeDefinition typeDefinition, BLangRecordTypeNode recordType,
+                                              String structName) {
         // Check if its an anonymous struct
         if (recordType.isAnonymous) {
             structName = "Anonymous Record " + structName.substring(structName.lastIndexOf('$') + 1);
         }
-        List<Field> fields = getFields(recordType, recordType.fields, recordType.symbol.documentation.attributes);
-        return new RecordDoc(structName, recordType.symbol.documentation.description, new ArrayList<>(), fields);
+        List<Field> fields = getFields(recordType, recordType.fields,
+                typeDefinition.getMarkdownDocumentationAttachment());
+        BLangMarkdownDocumentation documentationNode = typeDefinition.markdownDocumentationAttachment;
```

Comment:
Looks like we have two ways to get BLangMarkdownDocumentation node from typeDef. One is using a getter and other is accessing eh public variable. This looks like redundant. So lets make this one way.